src/test/ui/consts/ptr_comparisons.rs

```diff
@@ -0,0 +1,46 @@
+// compile-flags: --crate-type=lib
+
+#![feature(const_panic, core_intrinsics, const_raw_ptr_comparison, const_ptr_offset)]
+
+const FOO: &usize = &42;
+
+macro_rules! check {
+    (eq, $a:expr, $b:expr) => {
+        pub const _: () =
+            assert!(std::intrinsics::ptr_guaranteed_eq($a as *const u8, $b as *const u8));
+    };
+    (ne, $a:expr, $b:expr) => {
+        pub const _: () =
+            assert!(std::intrinsics::ptr_guaranteed_ne($a as *const u8, $b as *const u8));
+    };
+    (!eq, $a:expr, $b:expr) => {
+        pub const _: () =
+            assert!(!std::intrinsics::ptr_guaranteed_eq($a as *const u8, $b as *const u8));
+    };
+    (!ne, $a:expr, $b:expr) => {
+        pub const _: () =
+            assert!(!std::intrinsics::ptr_guaranteed_ne($a as *const u8, $b as *const u8));
+    };
+}
+
+check!(eq, 0, 0);
+check!(ne, 0, 1);
+check!(!eq, 0, 1);
+check!(!ne, 0, 0);
+check!(ne, FOO as *const _, 0);
+check!(!eq, FOO as *const _, 0);
+check!(!eq, FOO as *const _, FOO as *const _);
```

Comment:
This is a weird one -- and if you add reflexivity to `guaranteed_eq`, I think this should become an `eq` test.

Reply:
`FOO` being a static, we definitely want this to be equal at some point. If you insist on keeping this test, please add a comment saying that we do want behavior of this to change at some point.

Reply:
I left a comment. I want to keep the test, but I'd be fine with changing the impl to return true for identical `AllocId`s

Reply:
For now I'd be happy with a comment saying that we probably want to change behavior of that `check!` at some point.